docs/design/release_automation_design.md

```diff
@@ -0,0 +1,72 @@
+## Release Automation Process Design
+
+### Abstract
+
+The release automation process is an automated system that will handle Github
+releases. What this entails is tagging a release, ensure that the release blurb
+contains all the changes that are to go out, and potentially handle release
+artifacts.
+
+### Specification
+
+The release automation process will be triggered by a periodic AWS Lambda
+(Lambda) job that will signify whether or not regeneration of the SDK needs to
+occur and/or tag a Github release.
+
+Regeneration of the SDK will be determined by diffing the swagger.yaml files in
+the SDK and Firecracker. If there is a difference in the swagger files, this
+will regenerate the SDK and also tag a new release. In the event that there are
+no differences, the SDK will check some sort of notifier that may trigger a
+newly tagged release. An example notifier could be a certain change in the
+CHANGELOG.md file which would flag for a release in Amazon DynamoDB (DynamoDB).
+By storing a release flag in a database allows for multiple different ways for
+triggering a release and allows for swapping the type of release notifier.
+
+The newly generated SDK will need to be tested before deployment or before
+tagging a release, which will use the GitLab test runner to do the testing.
+This will require some modifications on how code is deployed, meaning a
+separate branch.  Once that branch has been committed to the release process
+would need to create a PR and poll for success of the GitLab test runner.
+
+### Deployment
+
+CDK and AWS CloudFormation templates are an option, but this writing this stack
+will be short work in the AWS SDK for Go. The release process stack can be
+deployed to a new stack by simply running its make command with the given set
+of AWS credentials for the new stack. This will setup the proper DynamoDB
+tables, Lambda functions, IAM roles, and any other resources needed.
+
+### Manual Release Process
+
+Ensure that no PRs, commits, or other releases are in the process during the
+release process. It should be communicated to the team that a release is
+occurring to prevent bad releases.
+
+Once it has been communicated that the release is about to being, the releaser
+must put on the release hat. You are now ready to start the release.
+
+1. Install release dependency if it is not already installed. `go get github.com/aktau/github-release`
+2. Clone the upstream repo. `git clone git@github.com:firecracker-microvm/firecracker-go-sdk.git`
+  1. If the upstream repo was already cloned, ensure master branch is rebased. If not, rebase.
```

Comment:
Rebasing is probably not something we want to do in the middle of a release.  I think cloning to a temporary location (like `/tmp`) would prevent the need for this and make it a little less dangerous.

We have a script internally that we use for ECS agent, ecs-init, and ECR credential helper releases that does this, along with making sure that we're releasing the exact SHA that we think we are.